vllm/sequence.py

```diff
@@ -259,6 +259,16 @@ def prompt_token_ids(self) -> List[int]:
         # We use the prompt of an arbitrary sequence.
         return next(iter(self.seqs_dict.values())).data.prompt_token_ids
 
+    def get_last_latency(self, now: float) -> float:
```

Comment:
@simon-mo I don't think what you are proposing would work. At the end of the batch generation this code, basically sets the last_toke_time to now and also gets how long time passed from the last batch generation.  This approach also captures extra latency in generation because of prefill. In my #2302 patch I tried to measure just the generation, iteration latency, which is interesting to me. Once this gets merged, I would try to rebase.

Reply:
+1 to @NikolaBorisov 

Additionally, the `append_token_id` is a method in the `sequence` not the `sequence_group`. All the latency timings are measured from the POV of the `sequence_group` (which I thought made sense given that is a 1:1 mapping from a user request).

I also think it is a good idea to have a single central "clock" in the `LLMEngine`